File: data/test00.java
/**
 * Module-level documentation for Coding Agent testing.
 */
package com.agent.test;

import java.util.Map;
import java.util.Optional;

/**
 * Manages user session and authentication.
 */
class UserManager {
    private final String dbUrl;

    public UserManager(String dbUrl) {
        this.dbUrl = dbUrl;
    }

    public Optional<Map<String, String>> getUserById(int userId) {
        // TODO: Implement database lookup
        return Optional.empty();
    }
}

public class Main {
    public static void main(String[] args) {
        // Java 14+ 新しい switch 式/パターンのテスト
        int status = 200;
        switch (status) {
            case 200, 201 -> System.out.println("Success");
            default -> System.out.println("Failure");
        }
    }
}

File: data/test00.cpp
/**
 * @file main.cpp
 * @brief Module-level documentation for Coding Agent testing.
 */
#include <cstdint>
#include <iostream>
#include <map>
#include <optional>
#include <string>
#define PI (3.14159265358979323846)
int main(void);
namespace test
{
static int g_var = 0;

class UserManager;

enum class Color
{
    Red,
    Blue
};

namespace
{
    class User
    {
    public:
        User() {}
    };
} // namespace

/**
 * @class UserManager
 * @brief Manages user session and authentication.
 */
class UserManager
{
private:
    std::string db_url;

public:
    struct Info
    {
        operator bool();
        Info& operator+=(const Info& other) { id += other.id; return *this; }
        int32_t id;
        std::string name;
    };

    explicit UserManager(const std::string& url)
        : db_url(url)
    {
    }

    std::optional<std::map<std::string, std::string>> get_user_by_id(int user_id)
    {
        // TODO: Implement database lookup
        return std::nullopt;
    }
};

UserManager::Info::operator bool()
{
    return 0 <= id;
}

int32_t operator+(UserManager::Info& a, UserManager::Info& b) { return a.id + b.id; }

} // namespace test

int main(void)
{
    using namespace test;
    UserManager userManager;
    int status = 200;
    switch(status) {
    case 200:
    case 201:
        std::cout << "Success" << std::endl;
        break;
    default:
        std::cout << "Failure" << std::endl;
        break;
    }
    return 0;
}
